dev/check_pyspark_custom_errors.py

```diff
@@ -0,0 +1,100 @@
+import sys
+
+sys.path.insert(0, "python")
+import os
+from pyspark import errors as pyspark_errors
+from pyspark.errors.exceptions import connect as pyspark_connect_errors
+
+
+def find_py_files(path, exclude_paths):
+    """
+    Find all .py files in a directory, excluding files in specified subdirectories.
+
+    Parameters
+    ----------
+    path : str
+        The base directory to search for .py files.
+    exclude_paths : list of str
+        A list of subdirectories to exclude from the search.
+
+    Returns
+    -------
+    list of str
+        A list of paths to .py files.
+    """
+    py_files = []
+    for root, dirs, files in os.walk(path):
+        if any(exclude_path in root for exclude_path in exclude_paths):
+            continue
+        for file in files:
+            if file.endswith(".py"):
+                py_files.append(os.path.join(root, file))
+    return py_files
+
+
+def check_errors_in_file(file_path, pyspark_error_list):
+    """
+    Check if a file uses PySpark-specific errors correctly.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the file to check.
+    pyspark_error_list : list of str
+        List of PySpark-specific error names.
+    """
+    errors_found = []
+    with open(file_path, "r") as file:
+        for line_num, line in enumerate(file, start=1):
+            if line.strip().startswith("raise"):
+                parts = line.split()
+                if len(parts) > 1 and parts[1][0].isupper():
```

Comment:
Can you add a comment which case this searches for?

Reply:
Sure. Added `# Check for 'raise' statement and ensure the error raised is a capitalized word.`